File: Python/551. Student Attendance Record I.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        a_count = 0
        l_count = 0
        for c in s:
            if c == 'A':
                l_count = 0
                a_count += 1
            elif c == 'L':
                l_count += 1
            else:
                l_count = 0
            if a_count > 1 or l_count > 2:
                return False
        return True

File: 2021/551. Student Attendance Record I.cpp
class Solution {
public:
    bool checkRecord(string s) {
        int absentCount = 0, lateCount = 0;
        for (int i = 0; i < s.length() && absentCount < 2 && lateCount < 3; i++) {
            if (s[i] == 'A') {
                absentCount++;
            }
            if (s[i] == 'L') {
                lateCount++;                
            } else {
                lateCount = 0;
            }
        }
        return absentCount < 2 && lateCount < 3;
    }
};